src/DateTimeInput/index.jsx

```diff
@@ -320,13 +320,25 @@ export default class DateTimeInput extends Component
 
     static getDerivedStateFromProps( props, state )
     {
+        let timestamp;
+
+        if ( props.value )
+        {
+            timestamp = props.value;
+        }
+        else if ( props.value === null )
+        {
+            timestamp = undefined;
+        }
+        else
+        {
+            timestamp = displayTimestamp( state.editingTimestamp, state.timestamp );
```

Comment:
Not strictly related to this issue but... while we’re in here:

That `displayTimestamp` function seems rather unnecessary 😄 It can be removed and this line can just be:
```suggestion
            timestamp = state.editingTimestamp || state.timestamp;
```